test/Yesod/Paginator/WidgetsSpec.hs

```diff
@@ -189,3 +188,21 @@ spec = withApp $ do
             , "<li class=\"next disabled\"><a>»</a></li>"
             , "</ul>"
             ]
+
+    describe "filterParams" $ it "works" $ do
+        let pageParamName = PageParamName "p"
+            pageNumber :: Int
+            pageNumber = 3
+            params :: [(Text, Text)]
+            params =
+                [ ("p", "2")
+                , ("p", "3")
+                , ("foo", "bar")
+                , ("ids[]", "1")
+                , ("ids[]", "2")
+                ]
+
+        assertEq
+            "filters page params not equal to the page number but keeps query params with the same name"
+            (filterParams pageParamName pageNumber params)
+            [("p", "3"), ("foo", "bar"), ("ids[]", "1"), ("ids[]", "2")]
```

Comment:
```suggestion
        filterParams pageParamName pageNumber params
          `shouldMatchList` [("p", "3"), ("foo", "bar"), ("ids[]", "1"), ("ids[]", "2")]
```

I'm fine with `shouldBe` too -- if you think preserving order is actually something we should ensure/assert.

Reply:
Yeah that's better, the `should*`assertions are a lot clearer. I mostly forget about them and use the ones in `yesod-test`.

When using the `should` assertions, I think we need to either lift those into IO, e.g. 
```
shouldBe' a b = liftIO (a `shouldBe` b)
```

An alternative would be to bring in the `should*` functions from (hspec-expectations-lifted)[https://hackage.haskell.org/package/hspec-expectations-lifted-0.10.0], but that adds a dependency. Is adding that dependency fine?

Hmm... that's a good point about whether to ensure the order is preserved, I think it probably should be.

Reply:
>  Is adding that dependency fine

I use that package in almost all my projects, so do feel free to bring it in. But since this is just a single assertion right now, if you want to use `liftIO` that's fine too.

Reply:
Cool, I added the dependency